src/System.Linq/src/System/Linq/ToCollection.cs

```diff
@@ -14,9 +14,18 @@ public static TSource[] ToArray<TSource>(this IEnumerable<TSource> source)
             {
                 throw Error.ArgumentNull(nameof(source));
             }
+            
+            if (source is TSource[] array)
+            {
+                return array;
+            }
```

Comment:
@VSaDov can correct me if I'm wrong, but I believe we explicitly decided against this as ToArray is meant to have copy semantics, and returning the same array will allow writes to the returned array to be visible on the original.